eventlet/wsgi.py

```diff
@@ -458,23 +458,26 @@ def handle_one_request(self):
                 self.close_connection = 1
                 return
 
-        headers_sent = []
-        self.environ = self.get_environ(headers_sent)
+        self.environ = self.get_environ()
         self.application = self.server.app
         try:
             self.server.outstanding_requests += 1
             try:
-                self.handle_one_response(headers_sent)
+                self.handle_one_response()
             except socket.error as e:
                 # Broken pipe, connection reset by peer
                 if support.get_errno(e) not in BROKEN_SOCK:
                     raise
         finally:
             self.server.outstanding_requests -= 1
 
-    def handle_one_response(self, headers_sent):
+    def handle_one_response(self):
         start = time.time()
         headers_set = []
+        headers_sent = []
+        # Push the headers-sent state into the Input so it won't send a
+        # 100 Continue response if we've already started a response.
+        self.environ['wsgi.input'].headers_sent = headers_sent
```

Comment:
it's a little tricky how this mutable/non-local becomes truthy below when we `headers_sent.append(1)` - I can see why you were trying to do the plumbing but agree the API breaking signature change was probably a mis-step.